### ISSUE TITLE
Couchbase randomly crashing during tests

### ISSUE THREAD

**Issue by Billuc** [couchbase/couchbase-lite-net#1372]
### Library Version

CouchbaseLite/3.0.0 - Build/124 - Commit/f733cb70

### .NET Runtime

.NET (Core) 5.0

### Operating System / Device Details

Ubuntu 20.04

### Expected behavior

Xunit tests should run smoothly and pass.

### Actual behavior

By running the tests multiple times, we get different results (no changes to the code or the environment were made by us in between). Either :  
1) Everything is fine and tests pass  
2) I get an exception when trying to create a database

```
Error Message:
   Couchbase.Lite.CouchbaseLiteException : CouchbaseLiteException (LiteCoreDomain / 11): can't open file.
  Stack Trace:
     at LiteCore.Interop.NativeHandler.ThrowOrHandle()
   at LiteCore.Interop.NativeHandler.Execute(C4TryLogicDelegate2 block)
   at LiteCore.LiteCoreBridge.Check(C4TryLogicDelegate2 block)
   at Couchbase.Lite.Database.<>c__DisplayClass101_0.<Open>b__0()
   at Couchbase.Lite.Support.ThreadSafety.DoLocked(Action a)
   at Couchbase.Lite.Database.Open()
   at Couchbase.Lite.Database..ctor(String name, DatabaseConfiguration configuration)
```

3) The tests crash and I get the following message : 

`The active test run was aborted. Reason: Test host process crashed : double free or corruption (fasttop)`

### Steps To Reproduce

1. Create a Xunit project
2. Create Xunit test classes and tests
3. In each test class constructor, create a new database : `new Database(Guid.NewGuid().ToString())`
4. Clear the database in the constructor and delete and dispose it in a Dispose method
5. Execute operations on the database in the tests (CRUD)
6. Run `dotnet tests` to launch all tests (multiple times to get the different results)

### Reproduction Project

I'll try to create one soon !

### Further comments

- I haven't been able to find a pattern in the way tests pass, crash or throw the exception
- There are 5 tests classes creating a new database with at least 5 tests in each (so there will be more than 25 databases created)
- Our tests were written to test repository classes which encapsulate the database, but the tests or the repositories don't appear to be the culprits (the exception seems related to Couchbase and the crash message to C/C++).

Do you have any clue as to why this is happening ? Tell me if there is more I can do to help
Billuc

**Comment by Sandychuang8** [comment]
.Net 5 is not officially supported. CBL issue is created to verify if the issue with the given steps here can be reproduced in .Net Core 3.1 and .Net 6.

**Comment by Billuc** [comment]
Hi,

Yes, sorry I should have provided that.
So, for step 4, we have `_database.Clear();` in the constructor.
And this is the Dispose method, that is called at the end of the tests : 
```C#
        public void Dispose()
        {
            _database.Delete();
            _database.Dispose();
        }
```

Is the call to Dispose too much ? May it be the reason for the problems we encounter ?

Thanks

**Comment by Sandychuang8** [comment]
Delete() will dispose the database. And I just noticed 
`There are 5 tests classes creating a new database with at least 5 tests in each (so there will be more than 25 databases created)`
and at this point I think I would like to see a repro project to determine the cause of the issue you are seeing.

**Comment by Billuc** [comment]
Ok thanks !
Yeah sure, I will create a project whenever I can :)

**Comment by Sandychuang8** [comment]
Close issue for now. Please feel free to reopen the issue when you have a repro case.

**Comment by Billuc** [comment]
Ok, I think I found what went wrong ! I haven't got the time to create a repro so I can't be 100% sure though.

As I explained earlier, each of our test creates a brand new database and deletes it after the test is finished.
While fiddling with my tests, I noticed that when the number of tests was quite small everything goes well, but once the number of tests running in parallel becomes greater than 32, I get the error mentioned above. It seems I can't have more than 32 couchbase databases alive at the same time (I may have missed it in the docs).

So to solve the issue, I configured Xunit to stop running tests in parallel and now everything is working properly.
